### ISSUE TITLE
Subcommand destroys ordering of options

### ISSUE THREAD

**Issue by laurentseal** [tj/commander.js#962]
Commander.js version: 2.13.0

The feature "subcommand in dedicated binary" destroys option ordering for boolean longopts. The bug is that, for unknown options, commander always assumes they have an argument. Boolean longopts don't have an argument, so commander assumes the next option to be an argument to the boolean option. Unknown options get sorted to the end of the argument list, so option ordering is nonsensical after the aforementioned wrong assumption has taken place.

Example:

```
$ ls
somecommand              somecommand-subcommand1
```

Contents of somecommand:
```
program.command('subcommand1', 'execute subcommand1').
```

Triggering the bug:
`$ ./somecommand subcommand1 --bool-option subcommand2 --subcommand2-option subcommand2-argument`

should result in 

`./somecommand-subcommand1 --bool-option subcommand2 --subcommand2-option subcommand2-argument`

being called but commander executes

`./somecommand-subcommand1 subcommand2-argument --bool-option subcommand2`

This is a serious bug because the usage output cannot be made correct for the user, it is confusing the user and it results in the program not working without user feedback in combination with bug #961 .
It also could lead to random logic being executed if the options are implemented in a specific manner in the subcommand.

**Comment by MatthewRalston** [comment]
Seconded. I'm having a similar issue with 'git-style subcommands'. I have to admit it's only happening for one of my subcommands. For example, I'm using a `--verbose` flag to enable debug level logging in a subcommand. Node v12.2.0, commander 2.20.0


## Expectation

I expect that the process.argv from the 'git-style subcommand' is identical in order to process.argv from the main command

## Observation

I observe a reordering of the process.argv array that interferes with positional argument parsing *AND* doesn't throw an error about non-flag arguments being passed/parsed. Perhaps this is not important, but the order of all flags/positional arguments is rearranged. Also, this is not being handled by 

## Steps taken to debug

Only affects one of my subcommands, 'probability' (sub2) which is correctly spelled in both, and no similar command or alias exists. Also, rearrangement of the command declarations in `mycommand` had no effect.

```bash
bin/mycommand sub1 --verbose positionalArg1 # works fine
bin/mycommand sub2 --verbose positionalArg1 positionalArg2 #Causes the process.argv order to be rearranged 
```

To find *where* the rearrangement is being done, I did the following


mycommand
```node
program.description(...)
   .version(...)
   .command('sub1 <positionalArg1> <positionalArg2>', 'sub1 does distance')
   .command('sub2 <positionalArg1> <positionalArg2>', 'sub2 does probability')

console.log(process.argv)
program.parse(process.argv)
```

mycommand-sub1
```node
program.description(...)
   .arguments('<positionalArg1> <positionalArg2>', 'these do something')
   .options('-v, --verbose', 'enable verbosity')
console.log(process.argv)
program.parse(process.argv)
```

```bash
>bin/mycommand sub2 --verbose positionalArg1 positionalArg2
[
   'path/to/node/v12.2.0/bin/node'
   'path/to/bin/mycommand'
   'sub2'
   '--verbose'
   'positionalArg1'
   'positionalArg2'
]
[
   'path/to/node/v12.2.0/bin/node'
   'path/to/bin/mycommand-sub2'
   'positionalArg2'
   '--verbose'
   'positionalArg1'
]
```

As you can see, the order is rearranged substantially from the original.

## Best guess as to where in the codebase?

The `commander.Command.prototype.normalize` function is invoked [here](https://github.com/tj/commander.js/blob/master/index.js#L471). From inspecting in the REPL, the normalize function returns the expected order. This result is passed to [`parseOptions`](https://github.com/tj/commander.js/blob/master/index.js#L700). This too seems to work as expected, and returns an object with an args attribute that looks as expected with known options removed. Then the args are assigned to the [instance](https://github.com/tj/commander.js/blob/master/index.js#L472) and unknowns are passed to [`parseArgs`](https://github.com/tj/commander.js/blob/master/index.js#L648), and eventually the first positional argument is correctly assumed to be the name of a subcommand and [`executeSubCommand`](https://github.com/tj/commander.js/blob/master/index.js#L488) is called, passing in the original argv, the parse args (which are so far correct), and the unknown options.  It's worth noting that parseOptions returns some positional arguments in the unknown attributes if some options are not declared!! THIS MATTERS because an option (and therefore an unknown positional argument) can be 'unknown' in the main executable and then be [concatenated](https://github.com/tj/commander.js/blob/master/index.js#L512) to the end of the positional args! This rearranges all argv passed to the child process with `spawn`.

**Comment by shadowspawn** [comment]
Thanks for the detail @laurentseal and @MatthewRalston.

Related open issues: #508 
Related open PR: #734

(As I noted in #734, I am surprised things are potentially this broken, which makes me wonder if there are some calling patterns which are ok.)

**Comment by shadowspawn** [comment]
I am going to close this in favour of #508, and add a note there about the good diagnosis and reproduction steps here.

Thank you  for your contributions.

**Comment by shadowspawn** [comment]
FYI, opened a PR to preserve argument order: #1138

**Comment by shadowspawn** [comment]
The argument order is fixed in the published pre-release of Commander v5.0.0 (#1163)
